### DISCUSSION TITLE
Do you have any way of giving back?

### DISCUSSION THREAD

**Discussion by alexmace** [tsightler/ring-mqtt#394] [General Discussion]
I was thinking this integration works so well and usefully for me that I should be giving back to the author in some way - do you have anything to enable this? Like GitHub Sponsorships, Amazon Wish List, etc etc

**Comment by tsightler** [comment]
I should probably put something in the FAQ or on the main page since it comes up from time to time, actually, far more often than I would have expected.

I decided early on that I would not accept any contributions to this project.  This is just a personal project that I started with the goal to learn more about Javascript and NodeJS and then, after it reached a somewhat useable state, I made the fateful decision to share it publicly, but I don't really have much time for it these days and I think, if I did accept contributions, that I will feel obligated to provide more support for it or at least use the contributions in some way toward the project, and, unfortunately, contributions would not impact the amount of time I have available to dedicate to this project.

I do appreciate that people find this project useful enough to contribute to, however, even if I did accept contributions, I would likely just donate anything I received.  Based on that, I've asked people to consider one of two options if they feel like donating to this project:

1) Donate instead to a charity or person that could really use the help and just drop me a note.  I've been surprised how many people have done exactly this and I find it inspiring that a small side project can somehow help people in a way completely different and unexpected, even if the overall contribution is quite small.  
2) Donate to another OSS project, perhaps even an upstream project on which ring-mqtt depends, for example, [ring-client-api](https://github.com/dgreif/ring) without which this project could not exist.
